cmd/docker-proxy/udp_proxy_linux.go

```diff
@@ -137,6 +164,35 @@ func (proxy *UDPProxy) Run() {
 	}
 }
 
+func readDestFromCmsg(oob []byte, ipVer ipVersion) (_ net.IP, err error) {
+	defer func() {
+		// In case of partial upgrade / downgrade, docker-proxy could read
+		// control messages from a socket which doesn't have the sockopt
+		// IP_PKTINFO enabled. In that case, the control message will be all-0
```

Comment:
We don't leave docker-proxy running when the daemon stops, it'll always be a new socket I think? (Maybe I missed the point?!)